Insolvency.CalculationsEngine.Redundancy.API/Infrastructure/Middlewares/Validators/HolidayCalculationRequestValidator.cs

```diff
@@ -69,12 +69,16 @@ private bool NoOverlappingPeriods(List<HolidayTakenNotPaidCalculationRequestMode
 
         private bool RP1DataPresent(HolidayCalculationRequestModel data)
         {
-            return data.Rp14aNotRequired ? data.Htnp.Count(x => x.InputSource == InputSource.Rp1) > 0 : true;
+            return data.Htnp.Count(x => x.InputSource == InputSource.Rp14a) == 0 ||
+                   data.Htnp.Count(x => x.InputSource == InputSource.Rp1) > 0 ||
+                   data.Rp1NotRequired;
```

Comment:
I know this is a revert, but might be worth putting the data.Rp1NotRequired bool check first, before the linq queries - if it's true then it should speed things up by short circuiting the OR checks so the linq stuff never gets instantiated.